index.js

```diff
@@ -38,11 +38,16 @@ module.exports = function (rules) {
     });
 
     return function (files, metalsmith, done) {
+        var globalMetadata = metalsmith.metadata();
+
         Object.keys(files).forEach(function (file) {
             var fileObject = files[file];
 
             matchers.forEach(function (rule) {
                 if ( rule.matcher.match(file) ) {
+                    if (typeof rule.metadata === 'function') {
```

Comment:
I would rather put that in the `setMetadata` function

Reply:
@dpobel moved to setMetadata as suggested